CMakeLists.txt

```diff
@@ -18,7 +18,10 @@ set(
     )
 endif(CLANG-TIDY)
 
-add_compile_options(-Wall -Wextra -fconcepts-diagnostics-depth=5)
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options(-Wall -Wextra -fconcepts-diagnostics-depth=5)
```

Comment:
Although `-fconcepts-diagnostics-depth` is `gcc` specific other flags for more strict diagnostics make sense for `clang`.
Lets keep it for `clang` as well. It would be great to be able to provide `-fconcepts-diagnostics-depth` as `cmake` parameter, but it can be done as a separate PR.